static/js/compose_actions.js

```diff
@@ -110,7 +110,7 @@ function clear_box() {
     compose_validate.clear_all_everyone_warnings();
     compose.clear_private_stream_alert();
     compose_validate.set_user_acknowledged_all_everyone_flag(undefined);
-
+    compose.clear_preview_area();
```

Comment:
I suspect the bug is that this line should be moved from `hide_box` to here, not simply added here. But I'm not sure.

Reply:
From what I've gathered, `hide_box` is fired when clicking on a new stream while the compose box is open, or no message box is open, but does not fire when the composition box is in preview mode. Removing the call to `compose.clear_preview_area` in `hide_box` has no ill effects that I can produce.